linux/opt/sage/bin/apply_name_tag.sh

```diff
@@ -14,7 +14,8 @@ then
   exit 1
 fi
 NAME=$(echo $PRODUCTS | jq '.ProvisionedProducts[0].Name')
-ACCESS_APPROVED_ROLEID=$(echo $PRODUCTS | jq '.ProvisionedProducts[0].AccessApprovedRoleId')
+PROVISIONED_BY_ARN=$(echo $PRODUCTS | jq -r '.ProvisionedProducts[0].UserArn')
+ACCESS_APPROVED_ROLEID=$(/usr/bin/aws --region $AWS_REGION iam get-role --role-name ${PROVISIONED_BY_ARN##*/} | jq -r '.Role.RoleId')
```

Comment:
PROVISIONED_BY_ARN is a `user` ARN.  So how are you getting a role name out of PROVISIONED_BY_ARN to pass to the iam get-role command?

Reply:
Its an assume-role ARN when someone uses oidc login. The role name is end of the ARN path and I have to do another query to get the roleid. 

It would *fail* with an IAM user ARN.

Reply:
ohh, i see. is it because the product is actually provisioned by the SC EndUsers role?  What does the assume-role ARN look like?

Reply:
Assumed role ARNS look like this: `arn:aws:sts::237179673806:assumed-role/ServiceCatalogEndusers/0123456`

Reply:
If you ` aws servicecatalog search-provisioned-products` you get `"UserArn": "arn:aws:sts::$acct:assumed-role/ServiceCatalogEndusers"`